vcs/gitcmd/repo.go

```diff
@@ -1195,8 +1223,45 @@ func makeGitSSHWrapper(privKey []byte) (sshWrapper, keyFile string, err error) {
 	return tmpFile, keyFile, nil
 }
 
+// makeGitPassHelper writes a GIT_ASKPASS helper that supplies password over stdout.
+// You should remove the passHelper after using it.
+func makeGitPassHelper(pass string) (passHelper string, err error) {
+	f, err := ioutil.TempFile("", "go-vcs-gitcmd-ask")
+	if err != nil {
+		return "", err
+	}
+	tmpFile := f.Name()
+	if err := f.Chmod(0500); err != nil {
+		os.Remove(tmpFile)
+		return "", err
+	}
+	if _, err := f.WriteString("#!/bin/sh\necho '" + pass + "'\n"); err != nil {
+		os.Remove(tmpFile)
+		return "", err
+	}
+	if err := f.Close(); err != nil {
+		os.Remove(tmpFile)
+		return "", err
+	}
+	return tmpFile, nil
+}
+
 // InsecureSkipCheckVerifySSH controls whether the client verifies the
 // SSH server's certificate or host key. If InsecureSkipCheckVerifySSH
 // is true, the program is susceptible to a man-in-the-middle
 // attack. This should only be used for testing.
 var InsecureSkipCheckVerifySSH bool
+
+// environ is a slice of strings representing the environment, in the form "key=value".
+type environ []string
+
+// unset unsets a single environment variable.
+func (e *environ) unset(key string) {
```

Comment:
I can't. I don't want to modify the _current_ environment (of the process running), I want to modify the environment used for for starting a new command. [`os/exec.Cmd.Env`](https://godoc.org/os/exec#Cmd.Env) type is `[]string`.